From d34569439d4789ef209a04df02f3d6b1bf8a4e47 Mon Sep 17 00:00:00 2001
From: Zhu Honglei <zhuhonglei1714@phytium.com.cn>
Date: Mon, 17 Jun 2024 19:33:09 +0800
Subject: [PATCH 116/150] dt-bindings: edac: phytium,pe220x-edac: Add Phytium
 Pe220x support

This patch document the DT bindings for Phytium Pe220x EDAC.

Signed-off-by: Zhu Honglei <zhuhonglei1714@phytium.com.cn>
Signed-off-by: Li Mingzhe <limingzhe1839@phytium.com.cn>
Signed-off-by: Wang Yinfeng <wangyinfeng@phytium.com.cn>
Change-Id: I49d59e2d123d06f27616f6a165a1271ff3d219b4
Signed-off-by: Andrew Powers-Holmes <aholmes@omnom.net>
---
 .../bindings/edac/phytium,pe220x-edac.yaml    | 45 +++++++++++++++++++
 MAINTAINERS                                   |  1 +
 2 files changed, 46 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/edac/phytium,pe220x-edac.yaml

diff --git a/Documentation/devicetree/bindings/edac/phytium,pe220x-edac.yaml b/Documentation/devicetree/bindings/edac/phytium,pe220x-edac.yaml
new file mode 100644
index 000000000000..f36941fe97fd
--- /dev/null
+++ b/Documentation/devicetree/bindings/edac/phytium,pe220x-edac.yaml
@@ -0,0 +1,45 @@
+# SPDX-License-Identifier: GPL-2.0-only OR BSD-2-Clause
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/edac/phytium,pe220x-edac.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Phytium Pe220x SoC EDAC node
+
+maintainers:
+  - Zhu Honglei <zhuhonglei1714@phytium.com.cn>
+  - Chen Baozi <chenbaozi@phytium.com.cn>
+
+description: |
+  EDAC node is defined to describe on-chip error detection and correction for
+  Phytium Pe220x SoC.
+
+properties:
+
+  compatible:
+    const: phytium,pe220x-edac
+
+  reg:
+    maxItems: 3
+
+  interrupts:
+    maxItems: 3
+
+required:
+  - compatible
+  - reg
+  - interrupts
+
+additionalProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/interrupt-controller/irq.h>
+    edac: edac@32b28000 {
+            compatible = "phytium,pe220x-edac";
+            reg = <0x0 0x32b28000 0x0 0x1000>,
+                  <0x0 0x31400000 0x0 0x1000>,
+                  <0x0 0x31401000 0x0 0x1000>;
+            interrupts = <GIC_SPI 0 IRQ_TYPE_LEVEL_HIGH>,
+                         <GIC_SPI 1 IRQ_TYPE_LEVEL_HIGH>;
+    };
diff --git a/MAINTAINERS b/MAINTAINERS
index 273dc24a64c3..f659dc05f81d 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -2530,6 +2530,7 @@ ARM/PHYTIUM SOC SUPPORT
 M:	Chen Baozi <chenbaozi@phytium.com.cn>
 S:	Maintained
 W:	https://www.phytium.com.cn
+F:	Documentation/devicetree/bindings/gpio/phytium,pe220x-edac.yaml
 F:	Documentation/devicetree/bindings/gpio/phytium,gpio.yaml
 F:	Documentation/devicetree/bindings/gpio/phytium,sgpio.yaml
 F:	Documentation/devicetree/bindings/gpu/phytium,dc.yaml
-- 
2.47.0

